### PR TITLE
PNT Docker-Compose setup

### PR DESCRIPTION
**Pull request by gunsmoke**
I am used to working with Docker and Docker-compose.

This is my current development setup.

Adding this because it might help others that used docker-compose / docker setup faster.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+COPY . .
+
+ARG GOOS=linux
+ENV GO111MODULE=on
+
+# Setup the cache directory
+RUN mkdir -p /root/.pegnet/
+COPY defaultconfig.ini /root/.pegnet/defaultconfig.ini
+
+RUN go run Initialization/InitalizeTheNetwork.go
```
Comment (PaulBernier) [inline Dockerfile]: will this work as at this point no EC are available?

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    depends_on:
+      - factomd
+
+  pegnet:
+    network_mode: host
+    container_name: pegnet_container
+    build: .
+    volumes:
+      - .:/go/src/github.com/pegnet/pegnet
+      - ./defaultconfig.ini:/root/.pegnet/defaultconfig.ini:ro
```
Comment (PaulBernier) [inline docker-compose.yml]: I am not a Docker expert so I may be not understanding correctly, but in your Dockerfile you already copied that config file, so is it necessary to link this file via a volume?

File: Docker.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
+# PegNet Docker
```
Comment (PaulBernier) [inline Docker.md]: Could you add a note on this file on how to edit the config file with this set up?

File: docker-compose.yml (hunk 2 above)
Comment (gunsmoke) [reply docker-compose.yml]: I do it so I can Edit the default config on the fly ( this allows you to not rebuild the image every time you wanna change that file)

File: Dockerfile (hunk 1 above)
Comment (gunsmoke) [reply Dockerfile]: This is not 100% necessary and will probably change in the future, this was my way of getting some stuff installed in the image.

File: docker-compose.yml (hunk 2 above)
Comment (PaulBernier) [reply docker-compose.yml]: Oh ok, so does it override the file copied in the Dockerfile?

File: Docker.md (hunk 3 above)
Comment (gunsmoke) [reply Docker.md]: editing the file should `defaultconfig.ini` should work, we can do it another way

Copy the `defaultconfig.ini` > `myconfig.ini`

and we can change the volume part to
```
volumes:
      - ./myconfig.ini:/root/.pegnet/defaultconfig.ini:ro
``` 

Maybe this is a better solution, what do you thing @PaulBernier ?

File: docker-compose.yml (hunk 2 above)
Comment (gunsmoke) [reply docker-compose.yml]: yes, Dockerfile generates the base Image, then docker-compose does magic so we do not need to keep rebuilding the image every change.

File: Docker.md (hunk 3 above)
Comment (PaulBernier) [reply Docker.md]: Yep I think it's better. And add a line in the doc explaining that the config file is linked to the host file and that will be neat!

Comment (gunsmoke) [reply Docker.md]: will do, (will also add this to .gitignore)

**Comment by gunsmoke** [comment]
@PaulBernier Rebased with new changes.

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
+version: '3.2'
+
+services:
+  factomd:
+    network_mode: host
+    container_name: factomd_container
+    image: factominc/factomd:latest
+    entrypoint: /go/bin/factomd
+    command: '--count=5 --net=alot+ --blktime=15 --faulttimeout=20000 --enablenet=false --network=LOCAL --startdelay=2'
```
Comment (PaulBernier) [inline docker-compose.yml]: Any reason to spawn 5 simulated nodes?
Also I don't remember what does `--net` and `--enablenet` do, what are they and why are they necessary here?

Comment (gunsmoke) [reply docker-compose.yml]: No particular reason, I was following this script https://github.com/FactomProject/factomd/blob/master/scripts/g25.sh

Please suggest what to change, I am not 100% sure if that is the best settings for local testing =)

Comment (PaulBernier) [reply docker-compose.yml]: Ok then I would keep it as simple as:
`--network=LOCAL --blktime=15`

Comment (gunsmoke) [reply docker-compose.yml]: Changed

**Pull request merged**